src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -26,17 +33,157 @@ const ActorInit Shot_Sun_InitVars = {
     (ActorFunc)ShotSun_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/ShotSun_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/ShotSun_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000020, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 30, 60, 0, { 0, 0, 0 } },
+};
+
+void ShotSun_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ShotSun* this = THIS;
+    s32 params;
+
+    // Translation: Ocarina secret occurrence
+    osSyncPrintf("%d ---- オカリナの秘密発生!!!!!!!!!!!!!\n", this->actor.params);
+    params = this->actor.params & 0xFF;
+    if (params == 0x40 || params == 0x41) {
+        this->unk_1A4[0] = 0;
+        this->actor.flags |= 0x10;
+        this->actor.flags |= 0x2000000;
+        this->actionFunc = func_80BADF0C;
+        this->actor.flags |= 0x8000000;
+    } else {
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        this->actionFunc = ShotSun_UpdateHyliaSun;
+        this->actor.flags &= ~1;
+    }
+}
+
+void ShotSun_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ShotSun* this = THIS;
+    s32 params = this->actor.params & 0xFF;
+
+    if (params != 0x40 && params != 0x41) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+void ShotSun_SpawnFairy(ShotSun* this, GlobalContext* globalCtx) {
+    s32 params = this->actor.params & 0xFF;
+    s32 fairyParams;
+
+    if (this->timer > 0) {
+        this->timer--;
+        return;
+    }
+
+    switch (params) {
+        case 0x40:
+            fairyParams = 7;
+            break;
+        case 0x41:
+            fairyParams = 7;
+            break;
+    }
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ELF, this->actor.initPosRot.pos.x,
+                this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z, 0, 0, 0, fairyParams);
```

Comment:
Probably a good idea to write a comment about UB here, since fairyParams is not always set to a value before being used here. The function is never used in a context where it would happen originally, but it could end up causing issues if people aren't aware of this while modding.

Reply:
I added a comment. Is it good?

Reply:
Looks alright to me. And it should be easy to find whenever we introduce a build flag to avoid UB.